### ISSUE TITLE
unable to startup minqlx server - library dependences

### ISSUE THREAD

**Issue by deesdeel** [MinoMino/minqlx#97]
Hi there,

unfortunately I'm not able to run the run_server_x64_minqlx.sh as it doesn't find my libpython3.5:

`./qzeroded.x64: error while loading shared libraries: libpython3.5m.so.1.0: cannot open shared object file: No such file or directory`

I located the file and modified the startscript, such that it loads the python lib into the LD_LIBRARY_PATH:

`locate libpython3.5m
/usr/local/lib/libpython3.5m.a
/usr/local/lib/python3.5/config-3.5m/libpython3.5m.a
/usr/src/Python-3.5.9/libpython3.5m.a

$ cat run_server_x64_minqlx.sh
#!/bin/bash
cd "$(dirname "$0")"
export LD_PRELOAD=$LD_PRELOAD:./minqlx.x64.so
LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/lib
echo $LD_LIBRARY_PATH
LD_LIBRARY_PATH=./linux64:$LD_LIBRARY_PATH
exec ./qzeroded.x64 "$@"`

however now it claims it doesn't find the libsteam_api.so anymore, which is just in the linux64 subdirectory:

`$ ./run_server_x64_minqlx.sh
:/usr/local/lib
./qzeroded.x64: error while loading shared libraries: libsteam_api.so: cannot open shared object file: No such file or directory`

no matter how I arrange the search paths (also tried absolute path to the linux64 subfolder), it won't find any one of the two libraries.

I'm using Ubuntu 20.04 Server and have python3.8 and python3.5 installed on my system:
`$ python3 -V
Python 3.8.2
$ python3.5 -V
Python 3.5.9`

**Comment by notallmighty** [comment]
It looks for me like you would have forgotten to compile minoqlx.  - make etc...
Have you build it in proper steamapps/xxxxxx/qlds dir? Because after steam instalation you need to install quake there, somewhere. And later you add minqlx. Or have you installed old ubu and installed game, and now you do upgrade of all these? 
I have seen it. I have forgotten back then it is posix.

**Comment by deesdeel** [comment]
Thanks for your answer. I tried both, compile minqlx myself aswell as use the .so files from an update package, but none of them started. I meanwhile was able to solve my problem by reinstalling python3.5 with the --enable-shared option. Suddenly everything works.

**Comment by em92** [comment]
@deesdeel so finally you reinstalled python 3.5 with  - - enable-shared, then you compiled minqlx and everything was fine?

Also by "reinstalling python" you actually mean "recompiling python"?

**Comment by em92** [comment]
Also why do you compile against python 3.5 and not 3.8?

**Comment by em92** [comment]
> Also why do you compile against python 3.5 and not 3.8?

Nevermind. Found your problem in irc channel.

As for now, to compile against python 3.8 you must do
```
git checkout 0.5
make clean && make
```

Edit1: Just released 0.5.3. Pull latest changes `git pull origin master` and `make clean && make`. No need to switch branch.

**Comment by deesdeel** [comment]
Hi, ok thank you!
To clarify again: in the beginning I only had Python3.8, I tried to install minqlx but depending on if I compiled it myself or used the binaries, I was getting different error messages (PyExc_ValueError in first case, libpython3.5 missing in the second case). googling the second error message for a solution someone recommended to install python 3.5 seperately, which I did, but that didn't help in first place. First after trying one million other things and two days of madness I tried to reconfigure python 3.5 with the --enabled-shared option (that I hadn't used the first time I installed it) und re make --altinstalled it. And it seems, that did the trick, afterwards everything started fine.

thanks for the tip about the new release.. i'm actually a bit afraid to try what you said now, as it directly violates the never change a running system rule :D
